api/turing/service/router_deployment_service.go

```diff
@@ -283,21 +269,6 @@ func (ds *deploymentService) UndeployRouterVersion(
 		}
 	}
 
-	// Delete experiment engine plugins server
-	if routerVersion.ExperimentEngine.PluginConfig != nil {
```

Comment:
Since we have existing deployments that use this component, I think we should still attempt to delete it during the uninstallation (but ignore any errors in case the resource is not found). Let's add a TODO in the code to remove this logic and we can do so after a few months.

Reply:
Ahh that's a very good point; ok I'll add this method back and do the necessary error handling for this :)

Reply:
I've changed the error handling a little and turned the `NewPluginsServerService` method into a simple function so we don't need to keep the method as part of the service builder interface. When we're ready to remove this deletion logic, we can just remove this code block as well as the file containing the `NewPluginsServerService` function.